src/main/app/src/pages/toteutus/ToteutusForm/HakeutumisTaiIlmoittautumistapaSection/HakeutumisTaiIlmoittautumistapaSection.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+
+import { transparentize } from 'polished';
+import { useTranslation } from 'react-i18next';
+import { Field } from 'redux-form';
+import styled from 'styled-components';
+
+import { Alert } from '#/src/components/Alert';
+import { FieldGroup } from '#/src/components/FieldGroup';
+import {
+  createFormFieldComponent,
+  FormFieldEditor,
+  FormFieldIntegerInput,
+} from '#/src/components/formFields';
+import { SegmentTab } from '#/src/components/SegmentTab';
+import { SegmentTabs } from '#/src/components/SegmentTabs';
+import { Box, Radio } from '#/src/components/virkailija';
+import { Hakeutumistapa } from '#/src/constants';
+import { useFieldValue, useSetFieldValue } from '#/src/hooks/form';
+import { getThemeProp } from '#/src/theme';
+import { getTestIdProps } from '#/src/utils';
+
+import HakeutumisTaiIlmoittautusmistapaFields from './HakeutumisTaiIlmoittautumistapaFields';
+import { HakukohteetKaytossaChoice } from './HakukohteetKaytossaChoice';
+
+export const StyledGrayRadio = styled(Radio)`
+  background-color: ${getThemeProp('colors.grayLighten6', transparentize(0.5))};
+  padding: 19px 10px;
+  margin-bottom: 10px;
+`;
+
+export const StyledBlueBox = styled(Box)`
+  background-color: ${getThemeProp('colors.blueLighten4', transparentize(0.7))};
+  padding: 30px;
+  margin-bottom: 20px;
+`;
+
+const HakutapaFormField = createFormFieldComponent(({ onChange, value }) => {
+  const { t } = useTranslation();
+  return (
+    <SegmentTabs value={value}>
+      <SegmentTab value={Hakeutumistapa.HAKEUTUMINEN} onClick={onChange}>
+        {t('toteutuslomake.hakuTapa.hakeutuminen')}
+      </SegmentTab>
+      <SegmentTab value={Hakeutumistapa.ILMOITTAUTUMINEN} onClick={onChange}>
+        {t('toteutuslomake.hakuTapa.ilmoittautuminen')}
+      </SegmentTab>
+    </SegmentTabs>
+  );
+});
+
+export const HakeutumisTaiIlmoittautumistapaSection = ({
+  name = 'hakeutumisTaiIlmoittautumistapa',
+  language,
+  hasHakukohdeAttached,
+}) => {
+  const { t } = useTranslation();
+  const hakuTapa = useFieldValue(`${name}.hakuTapa`);
+
+  const hakukohteetKaytossaValinta = useFieldValue(
+    `${name}.isHakukohteetKaytossa`
+  );
+
+  // aiemmin tallennettujen fallback:
+  // jos toteutuksella on kiinnitettynä hakukohteita, asetetaan isHakukohteetKaytossa-valinta arvoon kyllä
+  useSetFieldValue(
+    `${name}.isHakukohteetKaytossa`,
+    true,
+    hakukohteetKaytossaValinta !== true && hasHakukohdeAttached
+  );
+
+  const showHakeutumisTapaFieldsJaAloituspaikat =
+    hakukohteetKaytossaValinta === false;
+
+  const showHakukohteetKaytossaInfobox = hakukohteetKaytossaValinta === true;
```

Comment:
Jos hakukohteetKaytossaValinta ei voi olla null / undefined niin:

```suggestion
  const showHakukohteetKaytossaInfobox = hakukohteetKaytossaValinta;
```

Reply:
voi olla null/undefined